src/frontends/onnx/frontend/src/frontend.cpp

```diff
@@ -135,7 +154,42 @@ bool FrontEnd::supported_impl(const std::vector<ov::Any>& variants) const {
 void FrontEnd::add_extension(const std::shared_ptr<ov::Extension>& extension) {
     if (auto telemetry = std::dynamic_pointer_cast<TelemetryExtension>(extension)) {
         m_extensions.telemetry = telemetry;
+    } else if (auto transformation = std::dynamic_pointer_cast<DecoderTransformationExtension>(extension)) {
+        m_transformation_extensions.push_back(transformation);
+    } else if (const auto& so_ext = std::dynamic_pointer_cast<ov::detail::SOExtension>(extension)) {
+        add_extension(so_ext->extension());
+        m_other_extensions.push_back(so_ext);
+    } else if (auto common_conv_ext = std::dynamic_pointer_cast<ov::frontend::ConversionExtension>(extension)) {
+        m_conversion_extensions.push_back(common_conv_ext);
+        for (int i = 1; i < ngraph::onnx_import::OperatorsBridge::LATEST_SUPPORTED_ONNX_OPSET_VERSION; ++i)
+            ngraph::onnx_import::register_operator(common_conv_ext->get_op_type(),
+                                                   i,
+                                                   "",
+                                                   [=](const ngraph::onnx_import::Node& context) -> OutputVector {
+                                                       return common_conv_ext->get_converter()(NodeContext(context));
+                                                   });
+    } else if (const auto onnx_conv_ext = std::dynamic_pointer_cast<ConversionExtension>(extension)) {
+        m_conversion_extensions.push_back(onnx_conv_ext);
+        for (int i = 1; i < ngraph::onnx_import::OperatorsBridge::LATEST_SUPPORTED_ONNX_OPSET_VERSION; ++i)
+            ngraph::onnx_import::register_operator(onnx_conv_ext->get_op_type(),
+                                                   i,
+                                                   "",
+                                                   [=](const ngraph::onnx_import::Node& context) -> OutputVector {
+                                                       return onnx_conv_ext->get_converter()(NodeContext(context));
+                                                   });
     } else if (auto progress_reporter = std::dynamic_pointer_cast<ProgressReporterExtension>(extension)) {
         m_extensions.progress_reporter = progress_reporter;
     }
 }
+
+FrontEnd::~FrontEnd() {
+    // We should remove new added operations manually due to deadlock in python GIL (pybind11/gil.h)
+    // It looks like the issue occurs when we use static c++ objects to store wrapped objects,
+    // in our case OperatorsBridge is static (singleton), and it stores ConvertionExtension.
+    for (const auto& conv_ext : m_conversion_extensions) {
+        for (int i = 1; i < ngraph::onnx_import::OperatorsBridge::LATEST_SUPPORTED_ONNX_OPSET_VERSION; ++i) {
+            ngraph::onnx_import::unregister_operator(conv_ext->get_op_type(), i, "");
+        }
+    }
+    ngraph::onnx_import::OperatorsBridge::load_initial_state();
```

Comment:
@tomdol could you look at these changes?
the issue appears when we try to use FrontEnds in python, for example:

```
fem = FrontEndManager()
def load_1():
    front_end = fem.load_frontend_by_name("onnx")
    front_end.add_extension(ConversionExtension(/*some translator */)))
    front_end.convert() 
    // etc
    
def load_2():
    front_end = fem.load_frontend_by_name("onnx")
    front_end.convert() 
    // etc

load_1()  // FrontEnd ctor is called,  new op translator is added to OperatorsBridge map
// FrontEnd destructor is called, but OperatorsBridge is alive and contains new translator
load_2() // <-- we did the changes for previous FrontEnd, so we do not expect new FrontEnd is also affected, need to reload translators in OperatorsBridge to use them here with new FrontEnd
// FrontEnd destructor is called
// the program ends, OperatorsBridge destructor is called 
// if the map still contain some python objects, the deadlock appears, looks like python waits when the static object OperatorsBridge is destroyed, OperatorsBridge waits the program ends or something like that
```

Reply:
As discussed offline, the current solution is fine. Changing the OpsBridge to a non-static instance, per Frontend object is the way to go but can be implemented in a separate PR.